apps/include/opt.h

```diff
@@ -270,17 +270,22 @@
  */
 # define OPT_PROV_ENUM \
         OPT_PROV__FIRST=1600, \
-        OPT_PROV_PROVIDER, OPT_PROV_PROVIDER_PATH, \
+        OPT_PROV_PROVIDER, OPT_PROV_PROVIDER_PATH, OPT_PROV_LOAD_CONFIG, \
         OPT_PROV__LAST
 
+# define OPT_LIBCTX_OPTIONS \
+        { "use_libctx", OPT_LIBCTX, '-', "Create a library context to load providers into (use before provider options)" }
+
 # define OPT_PROV_OPTIONS \
         OPT_SECTION("Provider"), \
         { "provider_path", OPT_PROV_PROVIDER_PATH, 's', "Provider load path (must be before 'provider' argument if required)" }, \
+        { "provider_config", OPT_PROV_LOAD_CONFIG, 's', "Load provider using a configuration file" }, \
```

Comment:
We already have `-config`, why add another option that essentially does the same thing?

Reply:
I can have a look..

-config is not global either.   

And then it does this block copy in the apps that use it..
 if ((conf = load_config_file(configfile)) == NULL)
        goto end;
    if (configfile != default_config_file && !app_load_modules(conf))
        goto end;

Reply:
> -config is not global either.

Well, no it isn't.  We don't support global options, that's well known.

> And then it does this block copy in the apps that use it..
> if ((conf = load_config_file(configfile)) == NULL)
> goto end;
> if (configfile != default_config_file && !app_load_modules(conf))
> goto end;

Yes?  The app library isn't holy, it may be changed.  However, I wonder if `OPENSSL_CTX_load_config()` shouldn't be made to take a `CONF *` instead of a file name, and if there shouldn't be a `CONF_modules_load_with_libctx()` that it could use.  That would be much more on par with already existing functionality.

Reply:
Ok rewritten..